### PR TITLE
Fix square root primitive

### PR DESCRIPTION
**Pull request by bcarlet**
The implementation of `fp_sqrt` in the standard primitive library gives incorrect results when `INT_WIDTH` is odd.

### CODE REVIEW

File: primitives/math.sv [hunk 1]
```diff
@@ -10,13 +10,14 @@ module fp_sqrt #(
...
     logic [$clog2(ITERATIONS)-1:0] idx;
 
-    logic [WIDTH-1:0] x, x_next;
-    logic [WIDTH-1:0] quotient, quotient_next;
-    logic [WIDTH+1:0] acc, acc_next;
-    logic [WIDTH+1:0] tmp;
+    logic [EXT_WIDTH-1:0] x, x_next;
+    logic [EXT_WIDTH-1:0] quotient, quotient_next;
+    logic [EXT_WIDTH+1:0] acc, acc_next;
+    logic [EXT_WIDTH+1:0] tmp;
```
Comment (rachitnigam) [inline primitives/math.sv:20]: Could you summarize what the problem was here and how the changes fix them?

**Review by rachitnigam — APPROVED** [review]
Small question but otherwise looks good!

File: primitives/math.sv (hunk 1 above)
Comment (bcarlet) [reply primitives/math.sv:20]: For background, the existing implementation looks to be the same as the one described [here](https://projectf.io/posts/square-root-in-verilog/). The algorithm used there computes the result bit-by-bit by examining pairs of bits (grouped relative to the binary point), starting at the MSB.

The problem is that in the existing implementation, this most-significant pair was assumed to be `in[WIDTH-1:WIDTH-2]`. That assumption is only valid if `INT_WIDTH` is even; otherwise, there should be an implicit leading 0 in the most-significant pair. From the above link:

> `WIDTH` must be a multiple of two, for example, when working with seven binary digits, you must set the width to eight.

In the change, `EXT_WIDTH` represents this extended width, which is just `WIDTH` if `INT_WIDTH` is even, or `WIDTH + 1` if `INT_WIDTH` is odd. The sizes of the internal registers grow accordingly, and the input port is zero-extended.

Comment (sampsyo) [reply primitives/math.sv:20]: Wow; super interesting!! Maybe a quick comment to this effect in the code would be helpful? Like, just that last bit explanation about what `EXT_WIDTH` is and why we need it could be useful for posterity.

**Comment by sampsyo** [comment]
Looks great!! Thanks for adding more documentation!
